### PR TITLE
feat(playback-manager): add playback manager

### PR DESCRIPTION
**Pull request by MrTimscampi**
Adds a playback manager store and a PlayerManager component.

The video player is now managed through a `v-dialog` component, which can be minimized in a corner.

**Known issue: The `v-dialog` will sometimes prevent scrolling when maximized then minimized again. It seems completely random and I have no idea how to fix it currently.**

### CODE REVIEW

File: pages/genre/_itemId/index.vue [hunk 1]
```diff
@@ -8,18 +8,15 @@
           class="play-button ml-auto mr-2"
           color="primary"
           min-width="8em"
-          :disabled="isPlayable"
           depressed
           rounded
-          nuxt
-          :to="`./${genre.Id}/play`"
+          @click="play({ items: [genre] })"
```
[the remark this answers is not in the document]
Comment (MrTimscampi) [reply pages/genre/_itemId/index.vue:13]: The play() function takes the genre and gets all the playable items in it, so songs. Items contains albums.

File: components/PlayerManager.vue [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+    <v-hover v-slot="{ hover }">
+      <v-card class="player-card" width="100%">
+        <video-player v-if="isPlaying" />
+        <v-fade-transition>
+          <v-overlay v-show="hover && isMinimized" absolute>
+            <v-btn icon @click="toggleMinimized">
+              <v-icon>mdi-arrow-expand-all</v-icon>
+            </v-btn>
+          </v-overlay>
+        </v-fade-transition>
```
[the remark this answers is not in the document]
Comment (MrTimscampi) [reply components/PlayerManager.vue:22]: It's planned for the OSD pass :)

File: pages/genre/_itemId/index.vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (Camc314) [reply pages/genre/_itemId/index.vue:13]: Ah yeah, I saw that when I wasn't testing locally

File: components/PlayerManager.vue (hunk 2 above)
[the remark this answers is not in the document]
Comment (Camc314) [reply components/PlayerManager.vue:22]: Sounds good ;)

**Review by Camc314 — APPROVED** [review]

**Pull request merged**